src/store/mutations/mutations-model.ts

```diff
@@ -34,6 +36,26 @@ export const mutateCurrentDate = (state: any, currentDate: string) => {
   if (!state.stateModel.ignoreChanges) mutateHaveChanges(state, true)
 }
 
+export const mutateIsImmediate = (state: any, isImmediate: boolean) => {
+  state.stateModel.incorporationDateTime.isImmediate = isImmediate
+  if (!state.stateModel.ignoreChanges) mutateHaveChanges(state, true)
+}
+
+export const mutateIsFutureEffective = (state: any, isFutureEffective: boolean) => {
+  state.stateModel.incorporationDateTime.isFutureEffective = isFutureEffective
+  if (!state.stateModel.ignoreChanges) mutateHaveChanges(state, true)
+}
+
+export const mutateFutureEffectiveDate = (state: any, futureEffectiveDate: string) => {
+  state.stateModel.incorporationDateTime.futureEffectiveDate = futureEffectiveDate
+  if (!state.stateModel.ignoreChanges) mutateHaveChanges(state, true)
+}
+
+export const mutateIncorporationDateTimeValid = (state: any, incorporationDateTimeValid: boolean) => {
```

Comment:
mutate**Is**IncorporationDateTimeValid  ?

Reply:
Agree!